### PR TITLE
add support for 10.80+

### PR DESCRIPTION
**Pull request by Milice**

### CODE REVIEW

**Comment by ranisalt** [comment]
Boss!

But you're missing these:
https://github.com/otland/forgottenserver/blob/master/src/definitions.h#L27-L29

**Comment by Milice** [comment]
Yeah i know, i didn't change that though since there are still alot more code to actually fully support 10.80 with the premium shop and whatnot, and i don't have a clean enough code to pull with so i just kept it simple for now :3

File: src/protocollogin.cpp [hunk 1]
```diff
@@ -89,10 +89,12 @@ void ProtocolLogin::getCharacterList(const std::string& accountName, const std::
...
 	//Add premium days
-	if (g_config.getBoolean(ConfigManager::FREE_PREMIUM)) {
-		output->add<uint16_t>(0xFFFF); //client displays free premium
-	} else {
-		output->add<uint16_t>(account.premiumDays);
+	if (version >= 1080) {
+		output->addByte(g_config.getBoolean(ConfigManager::FREE_PREMIUM) || account.premiumDays > 0);
+		output->add<uint32_t>(g_config.getBoolean(ConfigManager::FREE_PREMIUM) ? 0 : (time(nullptr) + (account.premiumDays * 86400)));
+	}
+	else {
```
Comment (WibbenZ) [inline src/protocollogin.cpp]: Indentation

**Comment by WibbenZ** [comment]
So you decided to release it? ;D

File: src/protocollogin.cpp [hunk 2]
```diff
@@ -89,10 +89,12 @@ void ProtocolLogin::getCharacterList(const std::string& accountName, const std::
 	}
 
 	//Add premium days
-	if (g_config.getBoolean(ConfigManager::FREE_PREMIUM)) {
-		output->add<uint16_t>(0xFFFF); //client displays free premium
-	} else {
-		output->add<uint16_t>(account.premiumDays);
+	if (version >= 1080) {
+		output->addByte(g_config.getBoolean(ConfigManager::FREE_PREMIUM) || account.premiumDays > 0);
+		output->add<uint32_t>(g_config.getBoolean(ConfigManager::FREE_PREMIUM) ? 0 : (time(nullptr) + (account.premiumDays * 86400)));
```
Comment (djarek) [inline src/protocollogin.cpp]: You should use a portable way to get UNIX time. (time() is dependant on POSIX compliance). Use the C++ chrono library (as an example, check OTSYS_TIME()).

Comment (ranisalt) [inline src/protocollogin.cpp]: Can't he just use `OTSYS_TIME()` there?

Comment (djarek) [inline src/protocollogin.cpp]: It returns time in *milliseconds* from the UNIX epoch.

**Comment by Milice** [comment]
Maybe ur right about the OTSYS_TIME() but time(nullptr) is used in other files, for example bed.cpp, so it shouldn't really be a problem

**Comment by ranisalt** [comment]
@Milice but `time()` is an unspecified function, and it is not reliable. It may return different values whether you are running Linux, Windows, BSD or whatever. It can be used fine where you just need to get a delta of time, but not reliably to get the current time as it is implementation dependant, which is what you want.

That is what I understand, you can read at http://en.cppreference.com/w/c/chrono/time

**Comment by MaxyTibia** [comment]
@Milice, the character list works 100%, but after login, got this error:
Control.cpp 1517: unknow packet type during game (Type = 0)

**Comment by Milice** [comment]
Yeah, u need to add a string and i think a uint16_t in the first byte of protocolgame

**Comment by ranisalt** [comment]
Any news on this?

**Comment by HyphenMaster** [comment]
When I made this change in my source code and compiled it I get tibia debug when I try to login in my character. Anyone know what's going on ?

**Pull request closed without merging**